homeassistant/components/risco/sensor.py

```diff
@@ -73,7 +73,6 @@ async def async_added_to_hass(self):
         self.async_on_remove(
             self.coordinator.async_add_listener(self._refresh_from_coordinator)
         )
-        await self.coordinator.async_request_refresh()
```

Comment:
I don't think we need to remove this. But we normally refresh the coordinator in the init module after creating it before loading platforms that need it so they already have data in the cache.

Reply:
I wanted to avoid refreshing the coordinator before the binary sensor platform is ready.
I can move this to a `EVENT_HOMEASSISTANT_STARTED` listener if you think that's better than removing.

Reply:
I don't think that is a problem. Why do you think that? I'd move the refresh before any platform is loaded.

Reply:
Because if we refresh before the binary sensor platform is loaded, the `zone_entity_id` attribute is missing.

Reply:
It will potentially be missing on the first sensor state update regardless since we don't control the order of platform load.

Reply:
And I don't think this is a problem since we have a guard in the sensor platform for missing zone entity_id. It will populate on next refresh.

Reply:
It won't populate on next refresh - the coordinator only fetches new events. I think that in this case delaying the value by 60 seconds (and unless HA was down for a long time, in most cases no event happened while HA was down, so there shouldn't be a value anyway) is better than having partial info forever.

Reply:
Ok. If we want to refresh earlier we can await the platforms forwarding and refresh the coordinator after that. See an example of this pattern here:

https://github.com/home-assistant/core/blob/f771d8ff142a98a47eb25adf726e8771bf2bf36a/homeassistant/components/ozw/__init__.py#L261-L296

Reply:
when I try that the tests hang.

Reply:
My bad. I got it working